logstash-core/src/test/java/org/logstash/config/ir/EventConditionTest.java

```diff
@@ -0,0 +1,131 @@
+    private long runId;
+
+    @Before
+    public void beforeEach() {
+        runId = TEST_RUN.incrementAndGet();
+        EVENT_SINKS.put(runId, new LinkedTransferQueue<>());
+    }
+
+    @After
+    public void afterEach() {
+        EVENT_SINKS.remove(runId);
+    }
+
+    @Test
+    @SuppressWarnings("rawtypes")
+    public void testInclusionWithFieldInField() throws Exception {
+        final PipelineIR pipelineIR = ConfigCompiler.configToPipelineIR(
+                "input {mockinput{}} filter { " +
+                        "mockfilter {} } " +
+                        "output { " +
+                        "  if [left] in [right] { " +
+                        "    mockoutput{}" +
+                        "  } }",
+                false
+        );
+
+        // left list values never match
+        RubyEvent leftIsList = RubyEvent.newRubyEvent(RubyUtil.RUBY);
+        List listValues = Arrays.asList("foo", "bar", "baz");
+        leftIsList.getEvent().setField("left", listValues);
+        leftIsList.getEvent().setField("right", listValues);
+
+        // left map values never match
+        RubyEvent leftIsMap = RubyEvent.newRubyEvent(RubyUtil.RUBY);
+        Map mapValues = Collections.singletonMap("foo", "bar");
+        leftIsMap.getEvent().setField("left", mapValues);
+        leftIsMap.getEvent().setField("right", mapValues);
+
+        // left and right string values match when right.contains(left)
+        RubyEvent leftIsString1 = RubyEvent.newRubyEvent(RubyUtil.RUBY);
+        leftIsString1.getEvent().setField("left", "foo");
+        leftIsString1.getEvent().setField("right", "zfooz");
+        RubyEvent leftIsString2 = RubyEvent.newRubyEvent(RubyUtil.RUBY);
+        leftIsString2.getEvent().setField("left", "foo");
+        leftIsString2.getEvent().setField("right", "zzz");
+
+        // right list value matches when right.contains(left)
+        RubyEvent rightIsList1 = RubyEvent.newRubyEvent(RubyUtil.RUBY);
+        rightIsList1.getEvent().setField("left", "bar");
+        rightIsList1.getEvent().setField("right", listValues);
+        RubyEvent rightIsList2 = RubyEvent.newRubyEvent(RubyUtil.RUBY);
+        rightIsList2.getEvent().setField("left", "zzz");
+        rightIsList2.getEvent().setField("right", listValues);
+
+        // non-string values match when left == right
+        RubyEvent nonStringValue1 = RubyEvent.newRubyEvent(RubyUtil.RUBY);
+        nonStringValue1.getEvent().setField("left", 42L);
+        nonStringValue1.getEvent().setField("right", 42L);
+        RubyEvent nonStringValue2 = RubyEvent.newRubyEvent(RubyUtil.RUBY);
+        nonStringValue2.getEvent().setField("left", 42L);
+        nonStringValue2.getEvent().setField("right", 43L);
+
+        RubyArray inputBatch = RubyUtil.RUBY.newArray(leftIsList, leftIsMap, leftIsString1, leftIsString2,
+                rightIsList1, rightIsList2, nonStringValue1, nonStringValue2);
+
+        final Map<String, Supplier<IRubyObject>> filters = new HashMap<>();
```

Comment:
nitpick: why is `filters` constructed and populated manually, but the others constructed with `Collections#singletonMap`?